### ISSUE TITLE
Configuration copy blocks build output

### ISSUE THREAD

**Issue by ianxm** [paulsgcross/haxe-injection#24]
The `ConfigMacro.handleDirectory()` copies config files to a new directory named with `[cwd]/[Compiler.getOutput()]` but this is where the compiler is going to write the output, so the build fails with something like
```
Sys_error("main.hl: Is a directory")
```
How can config files be used such that they don't block the compiler output file?

**Comment by paulsgcross** [comment]
Looks like the output path macro will need to be more robustly implemented for targets such as Hashlink. I'll dig into a proper fix later on for these targets.

Unfortunately, I will not have access to a computer for a couple of days. Could you wrap the line that generates the output directory with a conditional flag to handle the sort of output Hashlink expects for its local files?

**Comment by ianxm** [comment]
Thanks for looking into it. There's no rush.

**Comment by paulsgcross** [comment]
So, I've made writing the config to the output location more robust. This now stops the build fail and doesn't generate a folder with the output name. Hashlink is able to read the file.

It would be fixed, however during the parsing of the Json on the Hashlink target, _some_ of the keys are parsed into (I assume) integers. This causes it to fail and I have no idea why it is doing this, unless you've encountered something similar?

I'll try and get this new issue fixed and then I can push it, along with some quality-of-life improvements, into a new minor ver.

**Comment by paulsgcross** [comment]
Welp, I ended up fixing it by changing how config builder works. Basically, it will just copy anything from your source directory to your output direction (now without this issue) and then allow you to resolve Jsons from it. However, it will need you to explicitly state the type you're assigning the Json to, i.e. `var config : MyConfig = builder.resolveJson('config.json')`.

Released it in ver 3.1.0 with a nice QOL feature to boot :)
